unic-locale-impl/src/extensions/transform.rs

```diff
@@ -50,15 +50,94 @@ fn is_language_subtag(t: &str) -> bool {
 }
 
 impl TransformExtensionList {
+    /// Returns `true` if there are no tfields and no tlang in
+    /// the `TransformExtensionList`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use unic_locale_impl::Locale;
+    ///
+    /// let mut lo: Locale = "en-US-t-es-AR".parse()
+    ///     .expect("Parsing failed.");
+    ///
+    /// assert_eq!(lo.extensions.transform.is_empty(), false);
+    /// ```
     pub fn is_empty(&self) -> bool {
         self.tlang.is_none() && self.tfields.is_empty()
     }
 
+    /// Sets tlang on the `TransformExtensionList`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use unic_locale_impl::Locale;
+    /// use unic_langid_impl::LanguageIdentifier;
+    ///
+    /// let mut lo: Locale = "en-US".parse()
+    ///     .expect("Parsing failed.");
+    ///
+    /// let tlang: LanguageIdentifier = "es-AR".parse()
+    ///     .expect("Parsing failed on tlang.");
+    ///
+    /// lo.extensions.transform.set_tlang(tlang)
+    ///     .expect("Setting tlang failed.");
+    ///
+    /// assert_eq!(lo.to_string(), "en-US-t-es-AR");
+    /// ```
     pub fn set_tlang(&mut self, tlang: LanguageIdentifier) -> Result<(), LocaleError> {
         self.tlang = Some(tlang);
         Ok(())
     }
 
+    /// Returns the tvalue of tfield in the `TransformExtensionList`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use unic_locale_impl::Locale;
+    ///
+    /// let mut lo: Locale = "en-US-t-k0-dvorak".parse()
+    ///     .expect("Parsing failed.");
+    ///
+    /// assert_eq!(lo.extensions.transform.get_tfield("k0")
+    ///                .expect("Getting tfield failed."),
+    ///            Some(vec!("dvorak")));
+    ///
+    /// // Here tfield with tkey "m0" is not available
+    /// assert_eq!(lo.extensions.transform.get_tfield("m0")
+    ///                .expect("Getting tfield failed."),
+    ///            None);
+    /// ```
+    pub fn get_tfield(&self, tkey: &str) -> Result<Option<Vec<&str>>, LocaleError> {
```

Comment:
Hmm, how hard would it be to return `Result<Option<impl ExactSizeIterator>>`? If very hard, I'm ok with going for a `Vec` now.

Reply:
But def `AsRef<[u8]>` on input.